File: test/react-native/features/fixtures/scenario-launcher/android/src/main/java/com/bugsnag/reactnative/test/scenarios/NativeStackHandledScenario.kt
package com.bugsnag.reactnative.test

import android.content.Context
import com.bugsnag.android.Bugsnag
import com.facebook.react.bridge.Promise

class NativeStackHandledScenario(context: Context): Scenario(context) {

    override fun run(promise: Promise) {
        promise.reject(generateException())
    }
}

File: test/react-native/features/fixtures/scenario-launcher/src/scenarios/NativeStackHandledScenario.js
import Scenario from './Scenario'
import Bugsnag from '@bugsnag/react-native'
import { NativeInterface } from '../lib/native'

export class NativeStackHandledScenario extends Scenario {
  async run () {
    try {
      await NativeInterface.runScenario('NativeStackHandledScenario')
    } catch (e) {
      Bugsnag.notify(e)
    }
  }
}
